### PR TITLE
ENH: Add additional options for pacf ols

### PR DESCRIPTION
**Pull request by bashtage**
Rebase #3862 and change to enhancement
Clean up docs for pacf functions
Add option to use through pacf()

### CODE REVIEW

File: statsmodels/tsa/stattools.py [hunk 1]
```diff
@@ -635,40 +646,79 @@ def pacf_burg(x, nlags=None, demean=True):
     return pacf, sigma2
 
 
-#NOTE: this is incorrect.
```
Comment (jbrockmendel) [inline statsmodels/tsa/stattools.py]: This comment has been a huge bother.  Very glad to see it go.

File: statsmodels/tsa/stattools.py [hunk 2]
```diff
@@ -635,40 +646,79 @@ def pacf_burg(x, nlags=None, demean=True):
...
+    """
+    n = len(x)
+    # Demean the series
+    x = x - np.mean(x)
+    # Create lags for multivariate OLS
+    pacf = np.empty(nlags + 1)
+    pacf[0] = 1.0
+    if efficient:
+        xlags, x0 = lagmat(x, nlags, original='sep')
+        xlags = add_constant(xlags)
```
Comment (jbrockmendel) [inline statsmodels/tsa/stattools.py]: It took me a minute to figure out why de-meaning on line 701 doesn't affect the outcome.  Maybe for the sake of future readers only de-mean in the not-efficient case?

File: statsmodels/tsa/stattools.py [hunk 3]
```diff
@@ -703,14 +754,22 @@ def pacf(x, nlags=40, method='ywunbiased', alpha=None):
...
+
     Notes
     -----
     This solves yule_walker equations or ols for each desired lag
     and contains currently duplicate calculations.
     """
 
-    if method == 'ols':
-        ret = pacf_ols(x, nlags=nlags)
+    if method.startswith('ols'):
```
Comment (jbrockmendel) [inline statsmodels/tsa/stattools.py]: Do we want to deny invalid strings starting with "ols"?

**Comment by jbrockmendel** [comment]
Couple comments, generally looks quite nice.

File: statsmodels/tsa/stattools.py (hunk 2 above)
Comment (bashtage) [reply statsmodels/tsa/stattools.py]: I figured it just keeps the code simpler, but moved it anyway.

File: statsmodels/tsa/stattools.py (hunk 3 above)
Comment (bashtage) [reply statsmodels/tsa/stattools.py]: I changed this

File: statsmodels/tsa/stattools.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/tsa/stattools.py]: It is tested to clearly correct

**Comment by josef-pkt** [comment]
question (I didn't compare with #3862)

Is the new version close to pacf_yd and pacf_ld?
Should there be a unit test that is is close to them?  (i.e. not as different as the old ols version)

(I read only roughly over the algorithm, old unit tests pass.)

**Comment by bashtage** [comment]
Ran a small MC using a range of AR, MA and ARMA models.

Ranks

```
                          Lowest bias  Top 3 Bias  Lowest RMSE  Top 3 RMSE
burg                               10          28            5          42
ols                                 7          25            0           7
ols-inefficient                     4          16            0           3
ols-unbisaed                        6          13            0           0
ols-unbiased-inefficient            4          10            0           0
yule-walker                         0           3            0           1
yule-walker-mle                     4          17           14          41
ld_unbiased                         1           5            0           0
ld_biased                           3          17           11          41

```

Suggests that this method may be reasonable if bias is the main concern. LD and YW unbiased are no good.

File: statsmodels/tsa/stattools.py [hunk 4]
```diff
@@ -708,25 +765,38 @@ def pacf(x, nlags=40, method='ywunbiased', alpha=None):
...
-    elif method in ['ywm', 'ywmle', 'yw_mle']:
+    elif method in ('ywm', 'ywmle', 'yw_mle'):
         ret = pacf_yw(x, nlags=nlags, method='mle')
-    elif method in ['ld', 'ldu', 'ldunbiase', 'ld_unbiased']:
+    elif method in ('ld', 'ldu', 'ldunbiased', 'ld_unbiased'):
         acv = acovf(x, unbiased=True, fft=False)
         ld_ = levinson_durbin(acv, nlags=nlags, isacov=True)
-        #print 'ld', ld_
         ret = ld_[2]
     # inconsistent naming with ywmle
```
Comment (jbrockmendel) [inline statsmodels/tsa/stattools.py]: add TODO/FIXME to this comment?

File: statsmodels/tsa/stattools.py [hunk 5]
```diff
@@ -708,25 +765,38 @@ def pacf(x, nlags=40, method='ywunbiased', alpha=None):
...
+        efficient = 'inefficient' not in method
+        unbiased = 'unbiased' in method
+        ret = pacf_ols(x, nlags=nlags, efficient=efficient, unbiased=unbiased)
+    elif method in ('yw', 'ywu', 'ywunbiased', 'yw_unbiased'):
         ret = pacf_yw(x, nlags=nlags, method='unbiased')
-    elif method in ['ywm', 'ywmle', 'yw_mle']:
+    elif method in ('ywm', 'ywmle', 'yw_mle'):
         ret = pacf_yw(x, nlags=nlags, method='mle')
-    elif method in ['ld', 'ldu', 'ldunbiase', 'ld_unbiased']:
+    elif method in ('ld', 'ldu', 'ldunbiased', 'ld_unbiased'):
```
Comment (jbrockmendel) [inline statsmodels/tsa/stattools.py]: Closes #5212.

**Pull request merged**